qiskit/algorithms/aux_ops_evaluator.py

```diff
@@ -197,5 +189,6 @@ def _compute_std_devs(
         # when `variances` is a single value equal to 0., our expectation value is exact and we
         # manually ensure the variances to be a list of the correct length
         variances = np.zeros(len(observables), dtype=float)
+    # TODO: this will crash if quantum_instance is a backend
```

Comment:
Looks like either the type annotation is just wrong, this code does not work with `Backend`

Reply:
That was the 'normal' thing done - it was wanted to support plain backend at the public API. Applications etc do something like this, which is what the CircuitSampler does above. 
```
        self._quantum_instance = (
            backend if isinstance(backend, QuantumInstance) else QuantumInstance(backend=backend)
```

Arguably something like this needs to be done somewhere for the code to work with a backend directly

Reply:
Would `eval_observables` function be a proper place to do that?

Reply:
If you changed this
```
    std_devs = _compute_std_devs(
        observables_expect_sampled, observables, expectation, quantum_instance
    )
```    
to get the quantum_instance from the CircuitSampler after its been set above then the circuit sampler will have wrapped it if it was a plain backend
```
    std_devs = _compute_std_devs(
        observables_expect_sampled, observables, expectation, sampler.quantum_instance
    )
```    
that would save replicating the wrapping code, but perhaps a comment there might help understand why thats done if you choose to do it like that. This code is deprecated though and will be removed in in a couple of releases around Oct timeframe.